### PR TITLE
[Triton][DSV4] Fix non-stage1 paged MQA logits for KV block size > 1

### PR DESCRIPTION
**Pull request by skysnow2001**
## Summary

  This change is similar to #4440, which added block-paged KV-cache indexing to the stage1 paged MQA logits kernel, but applies the same fix to the fused-reduction Triton path.

  The fused-reduction kernel reduces the per-head contributions directly into the output logits instead of storing the per-head intermediate results produced by the stage1 path.

The Gluon implementation was previously enabled by default with supported Triton versions, including on architectures unsupported by the Gluon kernel.

 Gluon is now enabled only on:
  - gfx942
  - gfx950
  - gfx1250

  Other architectures, including RDNA4, use the fused-reduction Triton implementation instead of reaching the Gluon architecture assertion.

  ## Technical details

  The corrected indexing maps each logical token to its logical block and its offset within that block:

  ```python
  logical_block_idx = logical_kv_idx // KVBlockSize
  block_offset = logical_kv_idx % KVBlockSize
  physical_block_idx = kv_indices[batch, logical_block_idx]
```

  FP8 values and FP32 scales are then loaded using both the physical block index and the in-block token offset.

  After the existing packed value/scale split, the wrapper reshapes the value and scale tensors using the configured KV block size and passes their block and token strides to the Triton kernel.

## GSM8K

Tested on 8xgfx1201:
 ```python
lm_eval \
    --model local-completions \
    --tasks gsm8k \
    --model_args "model=deepseek-v4-flash-0731,tokenizer=deepseek-ai/DeepSeek-V4-Flash,base_url=http://localhost:8000/v1/completions,num_concurrent=64,max_retries=3,tokenized_requests=False,trust_remote_code=True" \
    --batch_size auto
```

| Tasks | Version | Filter | n-shot | Metric |   | Value |   | Stderr |
|---|---:|---|---:|---|---|---:|---|---:|
| gsm8k | 3 | flexible-extract | 5 | exact_match | ↑ | 0.9522 | ± | 0.0059 |
|  |  | strict-match | 5 | exact_match | ↑ | 0.9522 | ± | 0.0059 |

### CODE REVIEW

**Comment by cagrikymk** [comment]
@skysnow2001 Can you provide some benchmark data to show there is no regression?
Just kernel level is also fine.

The diff is relatively small but just to make sure.

**Comment by skysnow2001** [comment]
@cagrikymk Ran the AITER benchmark `aiter/op_tests/op_benchmarks/triton/bench_deepgemm_attention.py --perf` with KV block size 1 before and after this change.

| Batch | `next_n` | KV length | Before TFLOPS | After TFLOPS | Delta |
|---:|---:|---:|---:|---:|---:|
| 1 | 2 | 16,384 | 23.924571 | 22.423204 | -6.28% |
| 1 | 2 | 32,768 | 18.320086 | 18.091940 | -1.25% |
| 1 | 2 | 65,536 | 17.325144 | 17.360287 | +0.20% |
| 2 | 2 | 16,384 | 20.678254 | 20.659364 | -0.09% |
| 2 | 2 | 32,768 | 20.022709 | 20.015067 | -0.04% |
| 2 | 2 | 65,536 | 18.383385 | 18.346752 | -0.20% |
| 4 | 2 | 16,384 | 20.044968 | 19.918592 | -0.63% |
| 4 | 2 | 32,768 | 21.143624 | 21.068673 | -0.35% |
| 4 | 2 | 65,536 | 20.440847 | 20.395127 | -0.22% |
| 1 | 1 | 65,536 | 14.347264 | 14.259356 | -0.61% |
| 2 | 1 | 65,536 | 14.658192 | 14.626336 | -0.22% |
| 4 | 1 | 65,536 | 17.975081 | 17.925527 | -0.28% |
| 8 | 1 | 65,536 | 19.588616 | 19.523410 | -0.33% |

**Review by cagrikymk — APPROVED** [review]

**Review by azaidy — APPROVED** [review]
LGTM!

**Review by valarLip — APPROVED** [review]
Approved, hold before e2e ci pass
